### PR TITLE
Prometheus/OpenMetrics relay

### PR DESCRIPTION
**Pull request by Wilm0r**
Wrote and debugged this over the last week. Should be pretty solid at least for (OS) weather sensors, allowed me to integrate weather info in existing Grafana dashboards and now I'm finally getting alerts when batteries run empty.

### CODE REVIEW

File: examples/rtl_433_prometheus_relay.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+import time
+
+import dateutil.parser
+import http.server
+
+
+class rtl_433(object):
+	# These fields are used to separate and tag unique sensors, instead of being exported
+	# as readings/values.
+	_ID_FIELDS = [
```
Comment (zuckschwerdt) [inline examples/rtl_433_prometheus_relay.py:17]: probably should include `type` and `subtype`

**Comment by zuckschwerdt** [comment]
Thanks! Have you considered using a network input (syslog) to decouple this and not rely on pipes? That way you can run each server (rtl_433, relay) on it's own and possibly under a supervisor.

File: examples/rtl_433_prometheus_relay.py (hunk 1 above)
Comment (Wilm0r) [reply examples/rtl_433_prometheus_relay.py:17]: Ah! Those were missing on my OS sensors, didn't know. Added.

**Comment by Wilm0r** [comment]
> Thanks! Have you considered using a network input (syslog) to decouple this and not rely on pipes? That way you can run each server (rtl_433, relay) on it's own and possibly under a supervisor.

I did see that this morning when looking at existing exporters, missed it so far. Looks like the inside of it is still just json so that should be an easy fix. Will update the PR soon.

**Comment by Wilm0r** [comment]
> > Thanks! Have you considered using a network input (syslog) to decouple this and not rely on pipes? That way you can run each server (rtl_433, relay) on it's own and possibly under a supervisor.
> 
> I did see that this morning when looking at existing exporters, missed it so far. Looks like the inside of it is still just json so that should be an easy fix. Will update the PR soon.

I've finished this change but will let it run for a day or so here first.

One thing by the way, I saw in at least one of the exporters:

    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)

Was this added under the assumption that automatic multiplexing will happen? Because what actually will happen in all SO_REUSEPORT implementations that I know of, is the packets randomly (or deterministically through port numbers perhaps?) will go to one of the listener sockets.

**Comment by zuckschwerdt** [comment]
Yes, "Syslog" is just the JSON you'd get on stdout with a fixed header (just to adopt an existing protocol).

I don't remember how that SO_REUSEPORT got there, likely left over from some early debugging. It's indeed a cargo-cult way to do things that can't work. Feel free to remove all instances with your PR.

**Comment by Wilm0r** [comment]
Pushed including `SO_REUSEPORT` cleanup.

**Pull request merged**